### PR TITLE
Use jaraco.collections for code re-use on CaseInsensitiveDict.

### PR DESCRIPTION
**Pull request by jaraco**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Feature and Bugfix and code removal.

* **What is the related issue number (starting with `#`)**

#1231

* **What is the current behavior?** (You can also link to an open issue here)

CherryPy has its own limited implementation of a Case-Insensitive Dict.

* **What is the new behavior (if this is a feature change)?**

I was reviewing #1633, I realized I'd seen this pattern before... and built a general-purpose class for its purpose... and released it as a library for others to consume. This approach removes CherryPy's bespoke implementation, which has the reported deficiency, and replaces it with the proven class found in jaraco.collections.

### CODE REVIEW

File: cherrypy/lib/httputil.py [hunk 1]
```diff
@@ -377,49 +379,16 @@ def parse_query_string(query_string, keep_blank_values=True, encoding='utf-8'):
...
-        try:
-            return self[key]
-        except KeyError:
-            self[key] = x
-            return x
-
-    def pop(self, key, default):
-        return dict.pop(self, str(key).title(), default)
+    @staticmethod
+    def transform_key(key):
```
[the remark this answers is not in the document]
Comment (webknjaz) [reply cherrypy/lib/httputil.py]: I think, we could invent better name: this one doesn't give an idea of what's going on here.

[the remark this answers is not in the document]
Comment (jaraco) [reply cherrypy/lib/httputil.py]: Feel free to file a ticket/PR with [jaraco.collections](/jaraco/jaraco.collections).

**Review by jaraco — COMMENTED** [review]
In light of #1671, this dependency is not allowed, so the functionality must be inlined before this PR can move forward in the current master.

**Comment by webknjaz** [comment]
This pull request fixes 1 alert - [view on lgtm.com](https://lgtm.com/projects/g/cherrypy/cherrypy/rev/pr-a68c3dacfb2a8387fc0ded68b86be220b8c36987)

**fixed alerts:**

* 1 for Non-callable called

---

*Comment posted by [lgtm.com](https://lgtm.com)*

**Pull request merged**